tensorflow/lite/micro/kernels/cmsis-nn/conv.cc

```diff
@@ -111,12 +110,59 @@ TfLiteStatus CalculateOpData(TfLiteContext* context, TfLiteNode* node,
 }
 
 void* Init(TfLiteContext* context, const char* buffer, size_t length) {
-  return nullptr;
+  void* raw;
+  context->AllocatePersistentBuffer(
+      context, sizeof(int), &raw);
+  return raw;
 }
 
 void Free(TfLiteContext* context, void* buffer) {}
 
 TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
+#if defined(__ARM_FEATURE_DSP)
+  OpData data;
+  int32_t buf_size;
+
+  auto* params = reinterpret_cast<TfLiteConvParams*>(node->builtin_data);
+
+  const TfLiteTensor* input = GetInput(context, node, kInputTensor);
+  const TfLiteTensor* filter = GetInput(context, node, kFilterTensor);
+  const TfLiteTensor* output = GetOutput(context, node, kOutputTensor);
+
+  RuntimeShape input_shape = GetTensorShape(input);
+
+  const int input_depth = input_shape.Dims(3);
+  const int input_width = input->dims->data[2];
+  const int input_height = input->dims->data[1];
+  const int filter_width = filter->dims->data[2];
+  const int filter_height = filter->dims->data[1];
+  const int output_width = output->dims->data[2];
+  const int output_height = output->dims->data[1];
+
+  int* buffer_idx = reinterpret_cast<int*>(node->user_data);
+
+  TF_LITE_ENSURE_STATUS(CalculateOpData(
+      context, node, params, input_width, input_height, filter_width,
+      filter_height, output_width, output_height, input->type, &data));
+
+  if (data.padding.width == 0 &&
+      data.padding.height == 0 && (input_depth % 4 == 0) &&
+      params->stride_width == 1 &&
+      params->stride_height == 1 && filter_width == 1 && filter_height == 1) {
+    buf_size = arm_convolve_1x1_s8_fast_get_buffer_size(input_depth);
+  }
+  else
+  {
+    buf_size = arm_convolve_s8_get_buffer_size(input_depth, filter_width, filter_height);
+  }
+
+  node->user_data = buffer_idx;
+  if (buf_size > 0)  {
```

Comment:
That's a good point. I should check if buf_size is 0.

Reply:
I think ideally we should just do 

return context->RequestScratchBufferInArena(context, buf_size, buffer_idx)

But let's submit it as it is for now. I'll change the code once I add the check. The only thing that I suggest is to check the return value.